### PR TITLE
fix #690 -- blob packing/unpacking of native python bool, int, float, and complex.

### LINKED ISSUE #690: Python integers turn into Numpy integers when fetched.
In my table exist a non-primary column of type `longblob` that stores dictionaries.  Inside the dictionary some of the values are Python integers. However, when fetched, these Python integers turn into Numpy integers. It would be helpful if the types are kept the same at insert and fetch time.

DataJoint version = 0.12.0

Below is a toy example:
``` Python
@schema
class MyTable(dj.Manual):
    definition = """
    prim_key: varchar(32)
    ---
    attr: longblob
    
    """

MyTable().insert1(dict(prim_key=0, attr=dict(a=1)))

print(type(MyTable().fetch1()['attr']['a']))

# the output is <class 'numpy.int64'>
```

### PR DESCRIPTION
**Pull request by dimitri-yatsenko**
… and complex

### CODE REVIEW

File: datajoint/blob.py [hunk 1]
```diff
@@ -251,6 +263,34 @@ def pack_recarray(self, array):
     def read_sparse_array(self):
         raise DataJointError('datajoint-python does not yet support sparse arrays. Issue (#590)')
 
+    def read_int(self):
+        return int(self.read_value('int64'))
+
+    @staticmethod
+    def pack_int(v):
+        return b"\x0a" + np.array(v, dtype='int64').tobytes()
```
Comment (guzman-raphael) [inline datajoint/blob.py]: Is there a reason why we did not utilize decimal packing here? Python `int` are essentially boundless (memory-dependent). I believe `decimal` packing would be a closer representation as the length would be encoded.

File: datajoint/blob.py [hunk 2]
```diff
@@ -251,6 +263,34 @@ def pack_recarray(self, array):
...
+    @staticmethod
+    def pack_complex(v):
+        return b"\x0c" + np.array(v, dtype='complex128').tobytes()
+
+    def read_float(self):
+        return float(self.read_value('float64'))
+
+    @staticmethod
+    def pack_float(v):
+        return b"\x0d" + np.array(v, dtype='float64').tobytes()
```
Comment (guzman-raphael) [inline datajoint/blob.py]: Is there a reason why we did not utilize decimal packing here? Python `float` have a precision of 53 bits which means we would be storing unnecessary additional data.

File: datajoint/blob.py [hunk 3]
```diff
@@ -251,6 +263,34 @@ def pack_recarray(self, array):
...
+    @staticmethod
+    def pack_bool(v):
+        return b"\x0b" + np.array(v, dtype='bool').tobytes()
+
+    def read_complex(self):
+        return complex(self.read_value('complex128'))
+
+    @staticmethod
+    def pack_complex(v):
+        return b"\x0c" + np.array(v, dtype='complex128').tobytes()
```
Comment (guzman-raphael) [inline datajoint/blob.py]: We could utilize `decimal` packing here for the same reasons as `float` below. Python seems to capture the first 53 bits for each the `real` part and the `complex` part.

Comment (dimitri-yatsenko) [reply datajoint/blob.py]: here Python is not doing anything special and just uses the standard IEEE 754 encoding.

File: datajoint/blob.py (hunk 2 above)
Comment (dimitri-yatsenko) [reply datajoint/blob.py]: ok, added encoding of unbounded integers.

File: datajoint/blob.py (hunk 1 above)
Comment (dimitri-yatsenko) [reply datajoint/blob.py]: modified to support unbounded `int`

**Comment by guzman-raphael** [comment]
@dimitri-yatsenko Can you update `datajoint-python/docs-parts/intro/Releases_lang1.rst`?

**Pull request merged**